### PR TITLE
beacon/light/request: simple request framework

### PR DESCRIPTION
**Pull request by zsfelfoldi**
This PR implements a general-purpose framework for sync mechanisms requesting data from multiple servers. It is intended to be used primarily by the beacon chain light client but might be used for other purposes too as it does nothing specific to the beacon chain. It implements timeouts, very simple rate limits and server selection, basically abstracting away the multiple server logic from the actual sync processes.

### CODE REVIEW

File: beacon/light/request/scheduler.go [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+			select {
+			case stop := <-s.stopCh:
+				close(stop)
+				return
+			case <-s.triggerCh:
+				break loop
+			case <-s.testWaitCh:
+			}
+		}
+	}
```
Comment (holiman) [inline beacon/light/request/scheduler.go:217]: Alternative without label. Matter of style, I guess, do as you wish
```suggestion
	for triggered := false; !triggered; {
		select {
		case stop := <-s.stopCh:
			close(stop)
			return
		case <-s.triggerCh:
			triggered = true
		case <-s.testWaitCh:
		}
	}
```

Comment (zsfelfoldi) [reply beacon/light/request/scheduler.go]: committed

**Comment by holiman** [comment]
Doesn't compile? 
```
?   	github.com/ethereum/go-ethereum/beacon/engine	[no test files]
ok  	github.com/ethereum/go-ethereum/beacon/light	0.185s
# github.com/ethereum/go-ethereum/beacon/light/request [github.com/ethereum/go-ethereum/beacon/light/request.test]
beacon\light\request\scheduler.go:220:46: syntax error: unexpected bool in argument list; possibly missing comma or )
FAIL	github.com/ethereum/go-ethereum/beacon/light/request [build failed]
?   	github.com/ethereum/go-ethereum/beacon/merkle	[no test files]
?   	github.com/ethereum/go-ethereum/beacon/params	[no test files]
```

**Comment by zsfelfoldi** [comment]
> Doesn't compile?

Indeed, last commit had a missing bracket, fixed.

**Comment by fjl** [comment]
Included in #25901

**Pull request closed without merging**